### PR TITLE
feat(content-types): add research areas - INNO-1994

### PR DESCRIPTION
**Pull request by kalinchernev**
# PR description

Nothing special to note, regular composition of existing components.

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [ ] `package.json` is up-to-date and `@ecl/[system]-base` is part of the dependencies
- [ ] I have checked the dependencies
- [ ] I have given the fractal status “ready” to my component
- [ ] I have declared `@define mycomponent` in the SCSS file
- [ ] I have specified `margin: 0;` on the CSS component
- [ ] I have provided tests
- [ ] I follow the naming guidelines
- [ ] the component supports composition
- [ ] there are no hardcoded strings (all content come from the context)
- [ ] I have filled the README.md file (at least a few lines)
- [ ] My component is listed in the root README
- [ ] My PR has the right label(s)
- [ ] The redirects in `src/website/src/routes/Redirects.jsx` are up-to-date

### CODE REVIEW

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+    )}
+    {template === 'harmonised-g2' && (
+      <PageHeaderHarmonised
+        {...pageHeaderContent}
+        breadcrumb={breadcrumbHarmonised2}
+        className="ecl-page-header-harmonised--group2"
+      />
+    )}
+
+    <div className="ecl-u-pv-3xl ecl-u-pv-lg-4xl">
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: Based on the mocks, padding should be `ecl-u-pv-xl ecl-u-pv-lg-3xl`

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+
+              <p className="ecl-u-type-paragraph ecl-u-type-color-black ecl-u-mt-m ecl-u-mb-none">
+                There are 5 cross-cutting policies in the Integrated Maritime
+                Policy where research and innovation plays a large part - blue
+                growth, marine data and knowledge, maritime spatial planning,
+                integrated maritime suveillance, and sea basin strategies.
+              </p>
+
+              <h2
+                className="ecl-u-type-heading-2 ecl-u-mb-l ecl-u-mt-lg-3xl ecl-u-type-color-black"
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: Based on mocks, padding bottom should be **m** on mobile: `ecl-u-mb-m ecl-u-mb-lg-l ecl-u-mt-lg-3xl` ?

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx [hunk 3]
```diff
@@ -0,0 +1,498 @@
...
+                Some examples
+              </h3>
+
+              <p className="ecl-u-type-paragraph ecl-u-type-color-black ecl-u-mt-none ecl-u-mb-none">
+                <UnorderedList className="ecl-u-mt-l ecl-u-type-color-black">
+                  <UnorderedListItem>
+                    the interaction between aquaculture and the marine
+                    environment and the resulting competition for space has been
+                    addressed in the{' '}
+                    <Link href="/example" variant="external" label="COEXIST" />{' '}
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: missing external link icon?

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx [hunk 4]
```diff
@@ -0,0 +1,498 @@
...
+                </UnorderedList>
+              </p>
+
+              <p className="ecl-u-type-paragraph ecl-u-type-color-black ecl-u-mb-none">
+                Read more about{' '}
+                <Link href="/example" label="maritime spatial planning" />.
+              </p>
+
+              <h2
+                className="ecl-u-type-heading-2 ecl-u-mb-l ecl-u-mt-lg-3xl ecl-u-type-color-black"
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: difficult to understand the approach of the margin headings on the mockup, the top h2 have a different margin of this one, I guess it is an error in the mockup.

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx [hunk 5]
```diff
@@ -0,0 +1,498 @@
...
+              </h2>
+
+              <p className="ecl-u-type-paragraph ecl-u-type-color-black ecl-u-mt-none ecl-u-mb-none">
+                Integrated maritime surveillance provides authorities interested
+                or active in maritime surveillance with ways to exchange
+                information and data. Sharing data makes surveillance cheaper
+                and more effective.
+              </p>
+
+              <h2 className="ecl-u-type-heading-2 ecl-u-mb-l ecl-u-mt-lg-3xl ecl-u-type-color-black">
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: seems to be an h3 in the mockup?

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx [hunk 6]
```diff
@@ -0,0 +1,498 @@
...
+                plan, and are recognised as a major achievements of both the
+                strategy and the action plan.
+              </p>
+              <p className="ecl-u-type-paragraph ecl-u-type-color-black ecl-u-mb-none">
+                <Link href="/example" label="More information" /> about Atlantic
+                Ocean sea basin strategies.
+              </p>
+
+              <SocialMediaShare
+                className="ecl-u-mv-3xl"
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: Should be only top margin here I think, as the main parent container already gives bottom padding to the footer: `ecl-u-mv-3xl`

same for research-area-hub page

File: src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx [hunk 7]
```diff
@@ -0,0 +1,1301 @@
...
+
+              <h2
+                className="ecl-u-type-heading-2 ecl-u-mb-l ecl-u-mt-lg-3xl ecl-u-type-color-black"
+                id="policy-focus"
+              >
+                Policy focus
+              </h2>
+
+              <UnorderedList variant="no-bullet">
+                <UnorderedListItem className="ecl-u-pv-m ecl-u-border-bottom ecl-u-border-color-grey-15">
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx:229]: gap between the heading `<h2>` and 2 columns list seems too high, `ecl-u-pb-m` on the 2 first column solve the problem.

File: src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx [hunk 8]
```diff
@@ -0,0 +1,1301 @@
...
+
+              <h2
+                className="ecl-u-type-heading-2 ecl-u-mb-l ecl-u-mt-lg-3xl ecl-u-type-color-black"
+                id="events"
+              >
+                Related events
+              </h2>
+
+              <article className="ecl-u-border-bottom ecl-u-border-color-grey-15 ecl-u-d-flex ecl-u-pv-m">
+                <div className="ecl-u-flex-grow-0 ecl-u-mr-m">
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx:897]: right margin should be xl `ecl-u-mr-m`

File: src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx [hunk 9]
```diff
@@ -0,0 +1,1301 @@
...
+                className="ecl-u-type-heading-2 ecl-u-mb-l ecl-u-mt-lg-3xl ecl-u-type-color-black"
+                id="contact"
+              >
+                Contact
+              </h2>
+
+              <div className="ecl-row ecl-u-mt-xs">
+                <div className="ecl-col-12 ecl-col-md-6">
+                  <article className="ecl-u-pv-m ecl-u-height-100 ecl-u-box-sizing-border">
+                    <h2>Research Enquiry Service</h2>
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx]: missing `ecl-u-type-heading-2` class

File: src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx [hunk 10]
```diff
@@ -0,0 +1,1301 @@
...
+              <h2
+                className="ecl-u-type-heading-2 ecl-u-mb-l ecl-u-mt-lg-3xl ecl-u-type-color-black"
+                id="contact"
+              >
+                Contact
+              </h2>
+
+              <div className="ecl-row ecl-u-mt-xs">
+                <div className="ecl-col-12 ecl-col-md-6">
+                  <article className="ecl-u-pv-m ecl-u-height-100 ecl-u-box-sizing-border">
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx]: gap between 'Contact' parent heading is too high, 
therefore I think we need to remove top margin here and use `ecl-u-pb-m` 
and also remove top margin for the next `<h2>` in column.

File: src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx [hunk 11]
```diff
@@ -0,0 +1,1301 @@
...
+                href="/example"
+                icon={{
+                  shape: 'ui--corner-arrow',
+                  size: 'xs',
+                  transform: 'rotate-90',
+                }}
+              />
+
+              <SocialMediaFollow
+                className="ecl-u-mt-l"
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx]: This box becomes full-width in the mobile mockups. Must be taken into account?

**Review by papegaill — CHANGES_REQUESTED** [review]
Here is my review on the markup, in general, it's difficult to understand the differences of margin on the heading of the mockup, some `<h3>` as different padding than other, same for `<h2>`. I fully understand to not set top margin on the first `<h2>` of the page, but for the rest it should be consistent. Ideally we should not set margin classes on heading except for the first  heading of the page. What do you think?

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx (hunk 3 above)
Comment (kalinchernev) [reply src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: Not sure why it's not showing, it has the `variant` set.

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx (hunk 5 above)
Comment (kalinchernev) [reply src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: Me not sure either ...

**Comment by kalinchernev** [comment]
@papegaill thanks for the review! I tried to follow what the picture shows, but not all info I could extract, some headings i'm unsure of and I don't have access to another source of this info than the image.

For the spacing, agree, it was hard for me to make sure it's correct, will follow-up on the general discussion in slack

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx (hunk 3 above)
Comment (emeryro) [reply src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: there is no 'external' variant is ECL2. Icon has to be added in the link.
But that could be an improvement...

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx (hunk 5 above)
Comment (emeryro) [reply src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: yes, seems like an h3 indeed (based on the font size and spacing)

File: src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx (hunk 7 above)
Comment (emeryro) [reply src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx:229]: I'd rather keep the padding and remove bottom margin on heading (see comment below)

File: src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx (hunk 11 above)
Comment (emeryro) [reply src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx]: No, we should stick to the component's display. If this is reported later as an issue, we will update the component, if agreeded by COMM

**Comment by emeryro** [comment]
good job here!
I added some comments too.

You can have a look at other content types like [call for tenders](https://ec.europa.eu/component-library/playground/ec/?path=/story/templates-pages--call-for-tenders) or [department](https://ec.europa.eu/component-library/playground/ec/?path=/story/templates-pages--department), as an example.
here is what I generally do when creating content types based on utilities:
- use mostly margin-top. So I remove margin below headers and last paragraphs for instance. That way, I can follow the spacing as reported on the mockups more easily.
- keep components/composition as it is defined. For instance, the 'content-item' composition is widely used on templates (basically almost every `<article>` tag is using this composition, or a derivate of it). So we should not alter its display (not removing padding for instance). Ideally the content-item should be a template that is used directly instead of having to copy/paste it every time (but that would be another task).
- don't hesitate to copy/paste elements from other templates (when they are not direclty usable components)

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx (hunk 1 above)
Comment (papegaill) [reply src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: This is not updated on both _Hub_ and _Spoke_ markup. Checking other CT like _Main policy_ or _Departement_ we have  `ecl-u-pv-xl ecl-u-pv-lg-3xl`, in the other hand we have `ecl-u-pv-3xl ecl-u-pv-lg-4xl` for _Funding programme_. This really needs to be aligned from CT to CT, this paddings should be the same for all pages.

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx (hunk 5 above)
Comment (papegaill) [reply src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: not updated to a `<h3>`

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx (hunk 6 above)
Comment (papegaill) [reply src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: Not updated, looking at other CT, the SocialMediaShare doesn't have bottom margin.

File: src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx (hunk 8 above)
Comment (papegaill) [reply src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx:897]: Not updated, but I put this as resolved as other CT have the same right margin on date_block.

File: src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx (hunk 10 above)
Comment (papegaill) [reply src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx]: looking more closely, this should be a `<h3>`

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx [hunk 12]
```diff
@@ -0,0 +1,512 @@
...
+                </UnorderedListItem>
+              </UnorderedList>
+
+              <p className="ecl-u-type-paragraph ecl-u-type-color-black ecl-u-mt-m ecl-u-mb-none">
+                Read more about{' '}
+                <Link href="/example" label="maritime spatial planning" />.
+              </p>
+
+              <h2
+                className="ecl-u-type-heading-2 ecl-u-type-color-black ecl-u-mt-l ecl-u-mb-none"
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: To be consistent this `<h2>` should have same top margin as top ones.

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx (hunk 6 above)
Comment (papegaill) [reply src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: Not updated, looking at other CT, the SocialMediaShare doesn't have bottom margin.

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx (hunk 1 above)
Comment (papegaill) [reply src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: This is not updated on both **_Hub_** and **_Spoke_** markup. Checking other CT like **_Main policy_** or **_Departement_** we have  `ecl-u-pv-xl ecl-u-pv-lg-3xl`, in the other hand we have `ecl-u-pv-3xl ecl-u-pv-lg-4xl` for **_Funding programme_**. This really needs to be aligned from CT to CT, this paddings should be the same for all pages.

File: src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx [hunk 13]
```diff
@@ -0,0 +1,1312 @@
...
+                        size: 'xs',
+                        transform: 'rotate-90',
+                      }}
+                    />
+                  </article>
+                </div>
+              </div>
+
+              <h2
+                className="ecl-u-type-heading-2 ecl-u-mb-none ecl-u-mt-l ecl-u-type-color-black"
```
Comment (papegaill) [inline src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx]: Should have same top margin as top ones `ecl-u-mt-lg-3xl`

**Review by papegaill — CHANGES_REQUESTED** [review]

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx (hunk 1 above)
Comment (kalinchernev) [reply src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: Changed in a way to be as the department, for consistency.

File: src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx (hunk 5 above)
Comment (kalinchernev) [reply src/systems/ec/implementations/react/templates/pages/research-area-spoke/src/ResearchAreaSpokePage.jsx]: hopefully better now

File: src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx (hunk 10 above)
Comment (kalinchernev) [reply src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx]: hopefully better now

File: src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx (hunk 13 above)
Comment (papegaill) [reply src/systems/ec/implementations/react/templates/pages/research-area-hub/src/ResearchAreaHubPage.jsx]: Only this update is missing? Should have same top margin as top ones ecl-u-mt-lg-3xl

**Review by papegaill — CHANGES_REQUESTED** [review]

**Review by papegaill — APPROVED** [review]

**Pull request merged**